### PR TITLE
Use "Common Changelog" format for Changelog

### LINKED ISSUE #554: Update CHANGELOG.md to follow Common Changelog conventions
In order to maximize the benefit of our `CHANGELOG.md`, I propose we update it to follow the [Common Changelog](https://common-changelog.org/) conventions. In conjunction, we should find an appropriate location in our other docs to indicate that we are following these conventions, perhaps in our Contributing Guide.

### PR DESCRIPTION
**Pull request by danielfromearth**
Resolves #554.

Use [Common Changelog](https://common-changelog.org).

For instance, this change:
- recategorizes entries into "Common Changelog" groups
- formats headings
- uses an imperative mood
- adds author tags to each entry

This PR also adds a few missing entries to older release sections.

----
📚 Documentation preview 📚: https://earthaccess--584.org.readthedocs.build/en/584/

### CODE REVIEW

**Comment by danielfromearth** [comment]
@jhkennedy, @mfisher87 — following up from a mention of this yesterday — any suggestions for linting of markdown in a CI pipeline?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,224 +2,256 @@
 
 ## [Unreleased]
 
-* Changes
+### Changed
```
Comment (chuckwondo) [inline CHANGELOG.md:5]: I'm nit-picking here, but for general markdown linting, let's follow headings with a single blank line.
```suggestion
### Changed

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,224 +2,256 @@
...
+- [#483](https://github.com/nsidc/earthaccess/issues/483): Now using
+    [Search After](https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html#search-after)
+    for collection and granule searches to support deep-paging through large
+    result sets.
+- [#421](https://github.com/nsidc/earthaccess/issues/421): Enable queries to
+    Earthdata User Acceptance Testing (UAT) system for authenticated accounts.
 
-  * [#421](https://github.com/nsidc/earthaccess/issues/421): Removed the
+### Removed
+- **Breaking**: [#421](https://github.com/nsidc/earthaccess/issues/421): Remove the
```
Comment (chuckwondo) [inline CHANGELOG.md:26]: Place `:` inside boldface and move reference to end

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,224 +2,256 @@
...
+     path is now created prior to direct S3 downloads, if it doesn't already
+     exist.
+
+### Added
+- [#483](https://github.com/nsidc/earthaccess/issues/483): Now using
+    [Search After](https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html#search-after)
+    for collection and granule searches to support deep-paging through large
+    result sets.
+- [#421](https://github.com/nsidc/earthaccess/issues/421): Enable queries to
+    Earthdata User Acceptance Testing (UAT) system for authenticated accounts.
```
Comment (chuckwondo) [inline CHANGELOG.md:23]: Move reference to ends

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,224 +2,256 @@
...
+- Return S3 data links by default when in region.
+
+## [0.6.0] - 2023-09-20
+
+### Added
+- earthaccess.get_s3fs_session() can use the results to find the right set of S3 credentials
+
+### Fixed
+- earthaccess.search_datasets() and earthaccess.search_data() can find restricted datasets #296
+- distributed serialization fixed for EarthAccessFile #301 and #276
```
Comment (chuckwondo) [inline CHANGELOG.md:112]: add links

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,224 +2,256 @@
...
+      `earthaccess.store._open_urls`, replacing the hard-coded value of 8.
+- Return S3 data links by default when in region.
+
+## [0.6.0] - 2023-09-20
+
+### Added
+- earthaccess.get_s3fs_session() can use the results to find the right set of S3 credentials
+
+### Fixed
+- earthaccess.search_datasets() and earthaccess.search_data() can find restricted datasets #296
```
Comment (chuckwondo) [inline CHANGELOG.md:111]: add link

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,224 +2,256 @@
...
-  * [#562](https://github.com/nsidc/earthaccess/issues/562): The destination
-     path is now created prior to direct S3 downloads, if it doesn't already
-     exist.
+### Added
+- Address #427 by adding parameters to collection query
+- Add user-agent in the request to track usage, closes #436
+
+### Fixed
+- Fix #439 by implementing more trusted domains in the SessionWithRedirection
+- Fix #438 by using an authenticated session for hits()
```
Comment (chuckwondo) [inline CHANGELOG.md:40]: add links

**Review by chuckwondo — CHANGES_REQUESTED** [review]

**Comment by danielfromearth** [comment]
Thanks @chuckwondo for spotting these, and I will incorporate those changes!  I'm actually still working through the changelog entry-by-entry, and I'm currently about half-way through. 

Another note to address (maybe there will be more as I get to more recent entries):  I don't see the following changes listed in the changelog. Perhaps another person can double-check whether they should be added...

I think they would go in the `[0.5.3] - 2023-08-01` section:
- https://github.com/nsidc/earthaccess/pull/247
- https://github.com/nsidc/earthaccess/pull/259
- https://github.com/nsidc/earthaccess/pull/260
- https://github.com/nsidc/earthaccess/pull/261
- https://github.com/nsidc/earthaccess/pull/263
- https://github.com/nsidc/earthaccess/pull/265

**Comment by danielfromearth** [comment]
Also noting these, which may be missing:

For `[0.6.0] - 2023-09-20`:
- https://github.com/nsidc/earthaccess/pull/294

For `[0.7.0] - 2023-10-31`:
- https://github.com/nsidc/earthaccess/pull/320

For `[0.8.0] - 2023-11-29`:
- https://github.com/nsidc/earthaccess/pull/340
- https://github.com/nsidc/earthaccess/pull/347

**Comment by danielfromearth** [comment]
pre-commit.ci autofix

**Comment by danielfromearth** [comment]
I added a couple of the missing entries mentioned above, but left out the ones that seemed like they would be uninteresting to anyone other than the developers.

I think this is ready to merge!  Was more work than I thought it would be to go through all the entries and cross-check them against the PR and issue lists on GitHub. I tried my best to capture who contributed to each PR/issue, and be as inclusive as possible, but if I missed someone, know that it wasn't intentional.  Hopefully this is much better now, and gives credit to all the effort people have been doing to keep this awesome tool going and improving.

**Comment by mfisher87** [comment]
@danielfromearth AMAZING work!!! I appreciate that this is very hard and time consuming. Also _worth it_ to show appreciation for everyone's work. I haven't had a chance to review this yet, but I'd like to make time for that in today's hack day and have a "merge party" :) Will you be there?

**Comment by danielfromearth** [comment]
Thanks @mfisher87  :) 
We have some ESIP Openscapes preparation time scheduled at the same time as the first half of the hack day, but I should be there for the second half of the hack day.

**Comment by mfisher87** [comment]
Awesome. I want to spend some time on integration tests today too so I'll try and get that out of the way first ;)

**Review by mfisher87 — DISMISSED** [review]

**Review by mfisher87 — APPROVED** [review]

**Comment by mfisher87** [comment]
Some discussion reflecting on the use of an "unreleased" section the way we do: https://github.com/vweevers/common-changelog/issues/13

**Pull request merged**